### PR TITLE
Treat normal using and static usings separately when adding imports.

### PR DESCRIPTION
**Pull request by JoeRobich**
Fixes https://github.com/dotnet/roslyn/issues/30811 and https://github.com/dotnet/roslyn/issues/30734

### CODE REVIEW

**Comment by JoeRobich** [comment]
@CyrusNajmabadi & @sharwell please take a look

File: src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs [hunk 1]
```diff
@@ -23,11 +23,13 @@ protected AbstractAddImportsService()
         protected abstract ImmutableArray<SyntaxNode> GetGlobalImports(Compilation compilation);
         protected abstract SyntaxList<TUsingOrAliasSyntax> GetUsingsAndAliases(SyntaxNode node);
         protected abstract SyntaxList<TExternSyntax> GetExterns(SyntaxNode node);
+        protected abstract bool IsStaticUsing(TUsingOrAliasSyntax usingOrAlias);
 
-        private bool IsUsing(TUsingOrAliasSyntax usingOrAlias) => GetAlias(usingOrAlias) == null;
+        private bool IsUsing(TUsingOrAliasSyntax usingOrAlias) => !IsAlias(usingOrAlias) && !IsStaticUsing(usingOrAlias);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs]: Can you rename this to something like "IsSimpleUsing" or "IsNormalUsing"?

File: src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs [hunk 2]
```diff
@@ -79,12 +81,18 @@ public SyntaxNode GetImportContainer(SyntaxNode root, SyntaxNode contextLocation
...
             {
                 case TExternSyntax e: return externContainer;
-                case TUsingOrAliasSyntax u: return IsAlias(u) ? aliasContainer : usingContainer;
+                case TUsingOrAliasSyntax u:
+                    if (IsAlias(u))
+                    {
+                        return aliasContainer;
+                    }
+
+                    return IsStaticUsing(u) ? staticUsingContainer : usingContainer;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs]: the mix of `if/ternary` is offputting to me :)  just pick one or the other.

**Comment by CyrusNajmabadi** [comment]
Bless you for doing this :)

File: src/EditorFeatures/CSharpTest/AddUsing/AddUsingTests.cs [hunk 3]
```diff
@@ -5048,5 +5048,133 @@ static class Extensions
...
+namespace N
+{
+    using SAction = System.Action;
+
+    class C
+    {
+        public List<int> F;
+    }
+}
+");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/AddUsing/AddUsingTests.cs]: test with nested namespcaes, where one namespace has normal usings, and one has static usings (and vice versa).

File: src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs (hunk 2 above)
Comment (JoeRobich) [reply src/Workspaces/Core/Portable/AddImports/AbstractAddImportsService.cs]: I have never found nested ternaries to be readable. =(

**Pull request merged**